arangod/RocksDBEngine/Methods/RocksDBTrxBaseMethods.cpp

```diff
@@ -106,7 +109,20 @@ TRI_voc_tick_t RocksDBTrxBaseMethods::lastOperationTick() const noexcept {
 /// @brief acquire a database snapshot if we do not yet have one.
 /// Returns true if a snapshot was acquired, otherwise false (i.e., if we
 /// already had a snapshot)
-bool RocksDBTrxBaseMethods::ensureSnapshot() { return false; }
+bool RocksDBTrxBaseMethods::ensureSnapshot() {
+  if (_rocksTransaction->GetSnapshot() == nullptr) {
+    _rocksTransaction->SetSnapshot();
+    _readOptions.snapshot = _rocksTransaction->GetSnapshot();
+    TRI_ASSERT(_readOptions.snapshot != nullptr);
+    // we at least are at this point
+    TRI_ASSERT(_db || _readOptions.snapshot);
+    _lastWrittenOperationTick = _readOptions.snapshot
+                                    ? _readOptions.snapshot->GetSequenceNumber()
+                                    : _db->GetLatestSequenceNumber();
```

Comment:
As `_readOptions.snapshot` is guaranteed to be a non-nullptr here, the ternary operator doesn't seem necessary anymore.